src/imitation/algorithms/adversarial.py

```diff
@@ -178,9 +178,7 @@ def _next_expert_batch(self) -> Mapping:
 
     @property
     def gen_batch_size(self) -> int:
-        # TODO(shwang): Is `n_steps` really the generator batch size? Isn't the batch
-        #  size actually `n_steps` * `n_vec`? I'm surprised that this works.
-        return self.gen_algo.n_steps
+        return self.gen_algo.n_steps * self.venv.num_envs
```

Comment:
Is there any guarantee that `self.venv` is the same `VecEnv` used in `self.gen_algo`? `gen_algo` is constructed by the user and they could feed something else in.

We could use `self.venv.get_env().num_envs` instead?

Reply:
Also, in general `n_steps` may not be defined and this semantically only really makes sense for `OnPolicyAlgorithm`, so I'd change the type signature in `AdversarialTrainer`, `GAIL` and `AIRL` accordingly. (Note off-policy algorithms will often take a single step, update the model, etc.)

Reply:
Did you mean to change the type signature to `OnPolicyAlgorithm` wherever applicable? I already did after we discussed the problem in #231.

https://github.com/HumanCompatibleAI/imitation/blob/ac4f4a50a928063477b2f9661312ed5d48c0b3a4/src/imitation/algorithms/adversarial.py#L453-L459

Reply:
>  Is there any guarantee that self.venv is the same VecEnv used in self.gen_algo? 

In the __init__ function we call:

Reply:
I missed the `set_env`, does seem less concerning, but still more fool-proof to access directly.